Filter/AbstractDateFilter.php

```diff
@@ -132,18 +134,25 @@ public function getDefaultOptions()
      */
     public function getRenderSettings()
     {
-        $name = 'sonata_type_filter_date';
+        $types = [
+            'sonata_type_filter_date' => DateType::class,
+            'sonata_type_filter_datetime' => DateTimeType::class,
+            'sonata_type_filter_date_range' => DateRangeType::class,
+            'sonata_type_filter_datetime_range' => DateTimeRangeType::class,
+        ];
+
+        $type = 'sonata_type_filter_date';
```

Comment:
🤔 the `sonata_type_filter` is useless now, and a bit confusing

Reply:
```
<?php
switch (true) {
    case $this->time && $this->range:
        $type = DateTimeRangeType::class;
        break;

    case $this->time && !$this->range:
        $type = DateTimeType::class;
        break;

    case !$this->time && $this->range:
        $type = DateRangeType::class;
        break;

    default:
        $type = DateType::class;
}
```

Please tell me sincerely what you think. I like that because you get rid of the
string array.

Reply:
Your code looks cleaner, but still violating SOLID principles. I didn't want to touch the original code actually. If I am going to change this, I will make it SOLID then ;)

Reply:
`switch` is a code smell for OCP violation, but only in the case when you cannot predict the size of the array. Here there are 2 options, which means 4 combinations. But maybe you were thinking about the case when people add more options?

Reply:
Yes, they can either add more options or new types, since this is allowed. The other thing is not nice with current code is that the inheritance with properties. 

> `switch` is a code smell for OCP violation

Indeed :D

Reply:
Oh ok then... curious to see the new solution :)

Reply:
@greg0ire can you please check, does it look better?